File: src/dlp/ast/sentences/Read.ts

import {Expression} from "../expressions/Expression";
import {AbstractLocatable} from "../AbstractLocatable";
import {Sentence} from "./Sentence";

export class Read extends AbstractLocatable implements Sentence {

    expressions: Expression[]

    constructor( expressions: Expression[], line: number, column: number){
        super(line, column);
        this.expressions = expressions;
    }
}

File: src/ast/statements/Read.java
package ast.statements;

import ast.AbstractLocatable;
import ast.expressions.Expression;
import java.util.List;

public class Read extends AbstractLocatable implements Statement {

    private List<Expression> expressions;

    public Read(List<Expression> expressions, int line, int column) {
        super(line, column);
        this.expressions = expressions;
    }

    public List<Expression> getExpressions() {
        return expressions;
    }
}
